src/main/java/bio/terra/cli/serialization/userfacing/input/referenced/UpdateReferencedBqDatasetParams.java

```diff
@@ -0,0 +1,108 @@
+package bio.terra.cli.serialization.userfacing.input.referenced;
+
+import bio.terra.cli.businessobject.resource.BqDataset;
+import bio.terra.cli.serialization.userfacing.input.UpdateResourceParams;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+import javax.annotation.Nullable;
+
+/**
+ * Parameters for updating a BigQuery dataset workspace referenced resource. This class is not
+ * currently user-facing, but could be exposed as a command input format in the future.
+ */
+@JsonTypeInfo(use = JsonTypeInfo.Id.CLASS, include = JsonTypeInfo.As.PROPERTY, property = "@class")
+@JsonDeserialize(builder = UpdateResourceParams.Builder.class)
+public class UpdateReferencedBqDatasetParams {
+  private final UpdateResourceParams resourceParams;
+  private final @Nullable String datasetId;
+  private final @Nullable String projectId;
+  private final BqDataset originalResource;
```

Comment:
The `serialization` package classes should all be POJOs, and not include classes from other CLI packages, with the  exception of enums. You can replace this with the individual properties that you need (e.g. `originalProjectId`)

ditto for the other classes that do this (`UpdateReferencedGcsObjectParams` etc.)